src/Particle/SoaDistanceTableAA.h

```diff
@@ -118,15 +118,22 @@ struct SoaDistanceTableAA: public DTD_BConds<T,D,SC>, public DistanceTableData
     return index;
   }
 
-  ///update the iat-th row for iat=[0,iat-1)
+  /** update the iat-th row for iat=[0,iat-1)
+   * into element iat of Distances and Displacements skiping element iat - 0
+   * doesn't really seem correct
+   * unless Distances[0] is original and Distances[1] is the proposed.
+   * in which case the brief above is just misinformation
+   * copying to Distances, Displacements iat or iat-1 both pass test_particles
+   * test_wavefunctions
```

Comment:
Non of the unit test covers this function.
The math needs iat instead of iat-1.
Only the lower triangle of the N by N(Npad) symmetric distance matrix is updated excluding the diagonal.
The correctness of the upper triangle is ignored.
Currently the distances is still N by Npad. Because some code path still uses the upper triangle which is correct when the full table is updated from scratch.
The displacements only stores the lower triangle with each row padded.
Eventually all the codepath should be fixed using only the lower triangle and distances should store only half.

Reply:
I think you must write a unit test showing this in code. I'm still not sure I understand you properly and the comment for update conveyed none of this at all.  That said this PR doesn't really deal with this and if it is currently correct it should behave the same.  The meat is in the other files.

This comment or something more "correct" should go here but even this is a huge improvement over the original.